app/controllers/rosters_controller.rb

```diff
@@ -12,16 +12,21 @@ def new
 
   def create
     @roster = Roster.new(identifier_name: params[:identifier_name])
-    @roster.save!
 
     add_identifiers_to_roster
 
-    @organization.roster = @roster
-    @organization.save!
+    if @roster.roster_entries.empty?
```

Comment:
This feels like something that should be part of the model, or a new creator. WDYT?